sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -543,6 +709,65 @@ graph LR
     Modifiable -->|Finalize| Executable
 ....
 
+==== Executable Graph Update
+
+A graph in the executable state can have the configuration of its nodes modified
+using a concept called graph _update_. This avoids a user having to rebuild and
+finalize a new executable graph when only the inputs & outputs to a graph
+change between submissions. Updates to a node take effect from the next
+submission of a graph and will not affect any previous submissions or in-flight
+executions of the same graph.
+
+The only type of node that is currently supported for updating in a graph is
+kernel execution nodes.
+
+The aspects of a kernel execution node that can be configured during update are:
+
+* Parameters to the kernel.
+* ND-Range of the kernel.
+
+Two methods are provided by the API to the user for performing this update.
+An API for updating the complete graph object, useful when the graph was
+created from a library, and an individual node update API.
```

Comment:
how to create an graph from a library with the explicit API,  can it be done with the existing API add(cfg)?

Reply:
The add API takes a [command-group-function object](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command-group-function-object), so if they library returned a `std::function` or instance of a callable type, e.g  `struct CGFunctor { void operator()(sycl::handler& h) const { /* code here */ }; };` then that could be passed straight into the add(cfd) API. This usage seems unlikely though, the queue recording API I think is a more natural fit for libraries.